plonky3/src/prover.rs

```diff
@@ -474,14 +474,17 @@ where
 
     // observe the parts of the proving key which correspond to the sizes of the tables we are proving
     if let Some(proving_key) = proving_key {
-        for commitment in proving_key
+        proving_key
             .preprocessed
             .iter()
-            .map(|(name, map)| &map[&multi_table.tables[name].degree].commitment)
-        {
-            challenger.observe(commitment.clone());
-        }
-    };
+            .filter_map(|(name, map)| {
+                multi_table
+                    .tables
+                    .get(name)
+                    .and_then(|table| map.get(&table.degree).map(|entry| &entry.commitment))
```

Comment:
```suggestion
                    .and_then(|table| map[&table.degree].commitment)
```

if the machine is there, this would skip the commitment if the size is not available. Instead, if the table is there, it's a hard error not to have a commitment for its size.

Reply:
Or a clean error that the machine cannot be proven because the setup doesn't cover it's length

Reply:
done